### DISCUSSION TITLE
[Projects-BETA] Hide specific status columns for different views

### DISCUSSION THREAD

**Discussion by COrtRey** [community/community#6989] [Projects and Issues] [answered]
Hi there!
My team would like to have two different board views to represent the two types of process we work on: discovery and delivery. Since each of those processes has different steps or statuses, we need to show different columns. 
I understand that you cannot create different columns in each view because the source of the content is the same and views are just a selection of the relevant elements. 
However, what should be possible is to hide some specific columns depending on the view. For example: "In Implementation" should only appear in the Delivery view and not in the Discovery view. I have looked everywhere and also tried with the filter feature but that only affects the issues/tickets and not the columns. 
Am I missing something or is that not available? This is a very basic feature and it would be a shame if you could not include it.
Thanks a lot in advance!

**Comment by serpro69** [comment]
Possible duplicate of https://github.com/github/feedback/discussions/6906 
Good to know I'm not the only one missing this feature :) Hope it will be implemented soon.

**Reply by COrtRey** [reply]
Thanks for reaching out, @serpro69. I am also glad I am not the only one who noticed it. I just upvoted your thread, let's hope we manage to get some visibility and they indeed implement it.

**Comment by BirgitPohl** [comment]
I just stumbled upon this, too.
I'm trying to filter out a column that is useful for another board but certainly a backlog, but it would be too much information for a specific development team:
<img width="1450" alt="Screenshot 2022-02-04 at 16 32 05" src="https://user-images.githubusercontent.com/4340405/152556496-8be52495-0929-4aa5-a2ad-a1a5e640132b.png">

Also, with filter status columns, I'd be able to integrate UX research and UI design into an agile circle. 
The Product Owner or project manager would be able to handle everything in one backlog and move the working items through the board, once an iteration of one team is finished.

**Comment by rafaelsales** [comment]
This would be awesome - my usecase is that our initial status of tickets is "Backlog", and we want to hide the backlog column from the kanban board. This is similar to the Jira Kanban backlog feature: https://support.atlassian.com/jira-software-cloud/docs/use-your-kanban-backlog/

**Comment by thiswayman** [comment]
I over-came this by adding: 

status:"status`` you want to include"

For multiple statuses I just separated each entry with a comma.

**Reply by crjones** [reply]
For you, does this remove the actual status columns or just the cards? For me it only removes the cards from those columns.

**Reply by yash-lysn** [reply]
Yeah apparently this works for some people as in it hides the actual column but for others it just hides the cards. I agree with this suggestion that having a show/hide toggle will probably be easiest here - https://github.com/github/feedback/discussions/6906#discussioncomment-2272405

**Comment by Norfeldt** [comment]
My current fix is to install an extension like stylebot and then do 

```css
[class*="column-frame"]:nth-of-type(3),
[class*="column-frame"]:nth-of-type(4) {
  display: none;
}
```

If there are several projects and you don't want the specific filter for all of them. then you have to narrow down the style URL to that specific project and have to do a refresh when changing project.

**Comment by amfio** [comment]
As this lack of functionally was one of the features preventing my team making the transition to the new GitHub projects, I have created a Chrome [extension](https://chrome.google.com/webstore/detail/github-projects-column-hi/mmaejmnikaleejkgkcjmabmmchejjlol) which looks for the `-status` filters and removes the columns which match that filter. 

For example, without the extension: the a filter of `-status:Backlog,Blocked` would remove all cards with a status of `Backlog` or `Blocked` but the board would still keep columns for those statuses even though they are guaranteed to be empty. All this extension does is remove those columns from the board.

Feel free to give it a go and see if it works for you and submit any [issues](https://github.com/amfio/github-projects-column-hider/issues) you have.

Alternatively you could grab the [source code](https://github.com/amfio/github-projects-column-hider/blob/master/src/script.js) and dump it in something like Tampermonkey 

Hopefully GitHub will add native support for this and I will remove the extension.

**Comment by 2percentsilk** [comment, marked as answer]
Check out our latest [release](https://github.blog/changelog/2022-09-02-the-new-github-issues-september-2nd-update/#%F0%9F%99%88-auto-hide-columns-with-board-filters) where we auto-hide columns based on your filter query! It should be available on all your Projects now 🥳

**Reply by michaelviveiros** [reply]
Very thanks!
